From 862518afc488a6bc00c1f8fae28e26e64f7f5197 Mon Sep 17 00:00:00 2001
From: Kees Cook <keescook@chromium.org>
Date: Wed, 19 Sep 2012 20:12:00 -0700
Subject: [PATCH] CHROMIUM: finit_module: add syscall to ARM

Add finit_module syscall to the ARM syscall list.

BUG=chromium-os:34134
TEST=snow build, manual testing

[accepted into Rusty's modules-wip tree for linux-next:
 http://git.kernel.org/?p=linux/kernel/git/rusty/linux.git;a=shortlog;h=refs/heads/modules-wip]
Change-Id: I7d35fd0cc98ff29e2a8dfae166d704d348908a2a
Signed-off-by: Kees Cook <keescook@chromium.org>
Reviewed-on: https://gerrit.chromium.org/gerrit/34305
Reviewed-by: Olof Johansson <olofj@chromium.org>
---
 arch/arm/include/asm/unistd.h | 2 ++
 arch/arm/kernel/calls.S       | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/arch/arm/include/asm/unistd.h b/arch/arm/include/asm/unistd.h
index 512cd14..eceefbe 100644
--- a/arch/arm/include/asm/unistd.h
+++ b/arch/arm/include/asm/unistd.h
@@ -404,6 +404,8 @@
 #define __NR_setns			(__NR_SYSCALL_BASE+375)
 #define __NR_process_vm_readv		(__NR_SYSCALL_BASE+376)
 #define __NR_process_vm_writev		(__NR_SYSCALL_BASE+377)
+					/* 378 for kcmp */
+#define __NR_finit_module		(__NR_SYSCALL_BASE+379)
 
 /*
  * The following SWIs are ARM private.
-- 
1.7.12.4

